sqjobs/brokers/multiqueue.py

```diff
@@ -18,19 +16,25 @@ def jobs(self, queue_names, timeout=0):
         * queue_names: List of queue names from where to read.
         * timeout: Not used by this implementation. It's kept for compatibility.
         """
-        queue_last_check = defaultdict(int)
+        queue_last_check = {queue: 0 for queue in queue_names}
 
         # loop through the queue list forever
-        for queue_name in cycle(queue_names):
-            next_check = queue_last_check[queue_name] + self.polling_interval
-
-            if next_check > int(time()):
-                sleep(next_check - int(time()))
-
-            # wait_time needs to be 0 or it will block in the first queue
-            payload = self.connector.dequeue(queue_name, wait_time=0)
-            if payload:
-                yield payload
-            else:
-                # only update last check when queue is empty
-                queue_last_check[queue_name] = int(time())
+        while True:
+            for queue in queue_names:
+                next_check = queue_last_check[queue] + self.polling_interval
+
+                # if it's been less than `polling_time` since last check, continue
+                if next_check > int(time()):
+                    continue
+
+                # wait_time needs to be 0 or it will block in the first queue
+                payload = self.connector.dequeue(queue, wait_time=0)
+                if payload:
+                    queue_last_check[queue] = 0
+                    yield payload
+                else:
+                    queue_last_check[queue] = int(time())
+
+            # only sleep if all queues are empty
+            if all(value != 0 for value in queue_last_check.values()):
```

Comment:
@diezguerra 
It will sleep if all the queues are empty.
Made me get rid of the `cycle()` and use a `while True` though :(